synapse/handlers/federation_event.py

```diff
@@ -897,6 +901,11 @@ async def _process_pulled_event(
                     backfilled=backfilled,
                 )
         except FederationError as e:
+            if backfilled:
+                await self._store.record_event_failed_pull_attempt(
+                    event.room_id, event_id, str(e)
+                )
```

Comment:
I'm not sure if this is the right place to record this? What is the failure mode you're trying to catch here?

Reply:
I assume this will catch the `FederationError` where `We can't get valid state history`,

<img width="1301" alt="" src="https://user-images.githubusercontent.com/558581/189935451-25a96093-0fd6-47f5-8cb2-cbf479fbe417.png">

---

But perhaps you also mean that we should also generally look for any exceptions and record there as well? That sounds reasonable for recording whatever failure that happens:

```py
try:
    ...
except FederationError as e:
    record_event_failed_pull_attempt(...)
    ...
except Exception as e:
    record_event_failed_pull_attempt(...)
    raise e
```

Reply:
Sorry, I was mostly just thinking that its not obvious that *this* particular spot is the right place in the handling. For example, I think by this point we've already attempted to try and refetch events that have bad signatures returned by `/backfill`.

I guess this is a good first step, and we can always add it the checks in more places later.